Android samsung 2.6.35

ytt3r opened the pull request:
I'm not sure why it said TeamKomin, its my development team. Interestingly enough, my global configs are set to ytt3r, so I'm not sure why it happened.

Please review my code and consider including it. I'll be changing some other files soon and sending you pull requests.

wildea01 committed:
ARM: 6387/1: errata: check primary part ID in proc-v7.S

Kconfig doesn't have any knowledge of specific v7 cores, so it is possible
to select errata workarounds that may cause inadvertent behaviour when
executed on a core other than those targetted by the fix.

This patch improves the variant and revision checking in proc-v7.S so
that the primary part number is also considered when applying errata
workarounds.

Acked-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Will Deacon <will.deacon@arm.com>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

Commit:
ARM: 6407/1: mmu: Setup MT_MEMORY and MT_MEMORY_NONCACHED L1 entries

This patch populates the L1 entries for MT_MEMORY and MT_MEMORY_NONCACHED
types so that at boot-up, we can map memories outside system memory
at page level granularity

Previously the mapping was limiting to section level, which creates
unnecessary additional mapping for which physical memory may not
present. On the newer ARM with speculation, this is dangerous and can
result in untraceable aborts.

Signed-off-by: Santosh Shilimkar <santosh.shilimkar@ti.com>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

wildea01 committed:
ARM: 6412/1: kprobes-decode: add support for MOVW instruction

The MOVW instruction moves a 16-bit immediate into the bottom halfword
of the destination register.

This patch ensures that kprobes leaves the 16-bit immediate intact, rather
than assume a 12-bit immediate and mask out the upper 4 bits.

Acked-by: Nicolas Pitre <nicolas.pitre@linaro.org>
Signed-off-by: Will Deacon <will.deacon@arm.com>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

Commit:
ARM: 6419/1: mmu: Fix MT_MEMORY and MT_MEMORY_NONCACHED pte flags

The commit f1a2481c0 sets up the default flags for MT_MEMORY and
MT_MEMORY_NONCACHED memory types. L_PTE_USER flag is wrongly
set as default for these entries so remove it. Also adding
the 'L_PTE_WRITE' flag so that these pages become read-write
instead of just being read-only

[this stops them being exposed to userspace, which is the main
concern here --rmk]

Reported-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Santosh Shilimkar <santosh.shilimkar@ti.com>
Acked-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

Commit:
[bcm4329] WOW support in BCM driver

Making sure that Rx pkt is acknowledged and sent
to network interface by wifi driver before system
goes in suspend.

BUG ID 767438

Change-Id: Ie88176ec11aa1d2fe4253d9b0472d13a4742d300
Reviewed-on: http://git-master/r/12703
Reviewed-by: Rakesh Kumar <krakesh@nvidia.com>
Reviewed-by: Bharat Nihalani <bnihalani@nvidia.com>
Tested-by: Bharat Nihalani <bnihalani@nvidia.com>

coolya commented:
looks goot, will merge asap i'm home

atinm commented:
Think this can be abandoned - the same changes were merged into atinm/android_kernel_samsung and then to coolya/android_kernel_samsung.

zachariasmaladroit committed:
jhash3 for 2.6.35 from dev.openwrt.org

zachariasmaladroit committed:
PM: Make it possible to avoid races between wakeup and system sleep
(upstream improvements) (Signed-off-by: Rafael J. Wysocki)

zachariasmaladroit committed:
add the Stochastic Fair Blue (SFB) network scheduler (by Juliusz
Chroboczek); Remove stub tcf function from sfb (by Jan Steffens)

Commit:
ashmem: Fix flush routine of ashmem

The ashmem flush function ashmem_flush_cache_range was calling
dmac_clean_range and dmac_flush_range with a user virtual address
which is not allowed. Instead of calling dmac_clean_range or
dmac_flush_range use flush_cache_user_range which can be called
with a user vitual address.

Change-Id: I5dd137fb248b45ef2c7ea5f7bdb4b540b8310af9
Signed-off-by: Shubhraprakash Das <sadas@codeaurora.org>

zachariasmaladroit committed:
BFQ v1-r1 for 2.6.35 by Fabio Checconi & Paolo Valente

Commit:
kbuild: Disable -Wunused-but-set-variable for gcc 4.6.0

[ upstream commit af0e5d565d2fffcd97d1e2d89669d627cc04e8b8 ]

Disable the new -Wunused-but-set-variable that was added in gcc 4.6.0
It produces more false positives than useful warnings.

This can still be enabled using W=1

[AK: dropped W=1 support in backport]
Signed-off-by: Dave Jones <davej@redhat.com>
Acked-by: Sam Ravnborg <sam@ravnborg.org>
Tested-by: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: Michal Marek <mmarek@suse.cz>
Signed-off-by: Andi Kleen <ak@linux.intel.com>

existz committed:
add Simple I/O Scheduler

michal42 committed:
kbuild: Fix passing -Wno-* options to gcc 4.4+

[ upstream commit 8417da6f2128008c431c7d130af6cd3d9079922e ]

Starting with 4.4, gcc will happily accept -Wno-<anything> in the
cc-option test and complain later when compiling a file that has some
other warning. This rather unexpected behavior is intentional as per
http://gcc.gnu.org/PR28322, so work around it by testing for support of
the opposite option (without the no-). Introduce a new Makefile function
cc-disable-warning that does this and update two uses of cc-option in
the toplevel Makefile.

Reported-and-tested-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Michal Marek <mmarek@suse.cz>
Signed-off-by: Andi Kleen <ak@linux.intel.com>

existz committed:
cpufreq: add interactiveX governor

jt1134 committed:
fix bml_over_mtd

Change-Id: Ieb8f51a89ab73a0cfcbcfd075cc2c1442c269024

jt1134 committed:
fascinate: fix mtd partition layout
fascinate has much smaller nand than epic does
match it to other sgs devices, with a nice healthy sized radio partition

Change-Id: I2c781149b8dbb488f9726bb12373b4eb77e7b0cf

jt1134 committed:
fascinate: add modemctl drivers from ED01
use cdma modemctl drivers, configurable with CONFIG_PHONE_ARIES_CDMA
modemctl gpio fixes so modem will actually try to init
needs mucho more work

Change-Id: I13c08e472a58a3341182eef794e9ed752694c170

jt1134 committed:
fascinate: add modemctl drivers from ED01
use cdma modemctl drivers, configurable with CONFIG_PHONE_ARIES_CDMA
modemctl gpio fixes so modem will actually try to init
needs mucho more work

Change-Id: I13c08e472a58a3341182eef794e9ed752694c170

atinm committed:
rename wm8994 'voodoo sound' extensions to just wm8994_extensions.

Change-Id: I99daa6e19d7f6cb0db64e192e2fd2115324c54ed

jt1134 committed:
fascinate: first round of gpio fixes

Change-Id: Id287eab340ade995727e6ac25d324f5f32c268fb

jt1134 committed:
Revert "fix bml_over_mtd"

This reverts commit 8aa8fd10b42063cca1c76e34fdf1725e8d73f6c2.

jt1134 committed:
Revert "fix bml_over_mtd"

This reverts commit 8aa8fd10b42063cca1c76e34fdf1725e8d73f6c2.

jt1134 committed:
remove unused files

Change-Id: I3d114a550d10ec681ffb5c3d85854910114e8bd1

jt1134 committed:
remove unused files

Change-Id: I3d114a550d10ec681ffb5c3d85854910114e8bd1

coolya committed:
jpeg: add new jpeg decoder settings from samsung

Change-Id: Ie4e5e9e913381d373f79a7ee1e319504a8d01337

coolya committed:
add DVFS_LIMIT for cpu-freq

Change-Id: I71b8b80c8c1fde8080d858c0e13e67108ba3cb83

coolya committed:
video: add new mfc driver from samsung source drop

Change-Id: I6efdadede02154f11c78d5179687b95fbe63e49e

coolya committed:
aries: config for DVFS_LIMIT

Change-Id: Icc6f3008393cc2dee2112145a973f0254cc5c862

coolya committed:
Merge branch 'android-samsung-2.6.35' of git://github.com/teamhacksung/android_kernel_samsung_aries into HEAD

coolya committed:
aries: add config for DVFS_LIMIT

coolya committed:
cpu-freq: fix missing define

rafl committed:
Fix build.sh shebang

atinm committed:
Merge pull request #28 from rafl/android-samsung-2.6.35

Fix build.sh shebang

atinm committed:
Merge pull request #27 from jt1134/android-samsung-2.6.35

Fascinate merge.

jt1134 committed:
DOH! fix bonehead typo

Change-Id: I7dc808f49cbe8ffd77bfaa27137f83abce30bee0

jt1134 committed:
fascinate: remove fm radio

Change-Id: Id74073a865d66051dd9a54f8772e738cad5c7ebd

jt1134 committed:
fascinate: additional gpio fixes

Change-Id: I9a1ba582e1f9e2eb97f36a897326063f614fc4c1

jt1134 committed:
fascinate: fix dpram driver initialization and add some debug output

Change-Id: Ibe06634cf0d4aca761954ac660a0cdb1b42bf5a1

jt1134 committed:
Merge branch 'android-samsung-2.6.35' of git://github.com/teamhacksung/android_kernel_samsung_aries into android-samsung-2.6.35

atinm committed:
turn off extension debug

Change-Id: Iaf8cd08535e9be21a0c4d1d7e938a7018fa289ca

atinm committed:
Allow lower gamma values for brightness (yellowish though, like herring)

Change-Id: Ic4b75bb874ecb31aa56bec330739bf6ea61c637e

jt1134 committed:
fascinate: apparently only dpram driver is needed for cdma radio
remove other modem drivers, and fix dpram irq settings I screwed up

Change-Id: Icf42e2942d18c10263c6617ac68be3679c319c3d

jt1134 committed:
fascinate: update defconfig
additional ppp options for cdma
kill ubernoisy debug in dpram

Change-Id: I8f46e0ed11f937498a2c891bac6268ffbf1830bd

coolya committed:
ce147: update driver from latest samsung source drop

Change-Id: Ia3fc856a7d321ee68058f0baf2229e4cfbab3860

coolya committed:
ce147: add missing autofocus mode

Change-Id: I6325ea7ced0efb77115a3e703508da91d2df35f0

coolya committed:
galaxys: add DIDLE cpu state

Change-Id: Ib0cd1f624eb8565ac414a49fb4050990f61b7307

jt1134 committed:
fascinate: clean-up. remove unused modem drivers

Change-Id: I3131862164349cbc504d652adc1ffe05c8025889

jt1134 committed:
Merge branch 'android-samsung-2.6.35' of git://github.com/teamhacksung/android_kernel_samsung_aries into android-samsung-2.6.35

jt1134 committed:
fascinate: shrink unused efs and radio partitions and donate the space to /cache and /system

Change-Id: I06c13697230cff5c7dd58bd90f0dfa2286e99d55

jt1134 committed:
fascinate: fix volume up button

Change-Id: Icfaba0f835cae6be4d6619c5d216ab7c63a89f88

jt1134 committed:
fascinate: drivers for kr3dh accelerometer

Change-Id: I47572a68f0106c260e7f719139ce3087c8c6d70d

jt1134 committed:
disable kr3dh debug

Change-Id: I2820ef250e55fe963b4d491545917fe392373dd0

jt1134 committed:
fascinate: fix UMS

Change-Id: I0162960c24f20e2ce1ce1c76f47b261552205cb1

Commit:
cpuidle: commit 7467571f4480b273007517b26297c07154c73924 upstream.

Cpuidle menu governor is using u32 as a temporary datatype for storing
nanosecond values which wrap around at 4.294 seconds. This causes errors
in predicted sleep times resulting in higher than should be C state
selection and increased power consumption. This also breaks cpuidle
state residency statistics.

atinm committed:
Merge pull request #29 from jt1134/android-samsung-2.6.35

DOH! fix bonehead typo

jt1134 committed:
fascinate: fix in-call audio

jt1134 committed:
fascinate: fix in-call audio

renaudallard committed:
Upgrade to 2.6.35.8

renaudallard committed:
Upgrade to 2.6.35.9

renaudallard committed:
Upgrade to 2.6.35.11

renaudallard committed:
Upgrade to 2.6.35.12

renaudallard committed:
Upgrade to 2.6.35.13

atinm committed:
Turn off LOG_INFO

Change-Id: Ieec151c3be8a6511299c6bc49c1973b9ccae3f91

coolya committed:
Merge pull request #30 from zachariasmaladroit/wifi

Wifi

Commit:
ashmem: Add cache flush routines to ashmem

coolya committed:
Revert "galaxys: add DIDLE cpu state"

This reverts commit 9213a898ee63b8f444d0504be5b12ff0cb3d1272.

coolya committed:
Merge branch 'android-samsung-2.6.35' of https://github.com/teamhacksung/android_kernel_samsung_aries into HEAD

existz committed:
add support for GPU overclock from TheEscapist

jt1134 committed:
fascinate: fix some gpio settings I screwed up

Change-Id: I299ceeabd5a712bcdc75223d5a6c15b18e411556

coolya committed:
aries: fix sleep of death

Change-Id: Ib2ec6067df3bf81de18ba938b98b1d6ae81a2eaa

LorDClockaN committed:
FIX: ext4: skip orphan cleanup if fs has unknown ROCOMPAT features

    commit d39195c33bb1b5fdcb0f416e8a0b34bfdb07a027 upstream.

    Orphan cleanup is currently executed even if the file system has some
    number of unknown ROCOMPAT features, which deletes inodes and frees
    blocks, which could be very bad for some RO_COMPAT features,
    especially the SNAPSHOT feature.

    This patch skips the orphan cleanup if it contains readonly compatible
    features not known by this ext4 implementation, which would prevent
    the fs from being mounted (or remounted) readwrite.

    Signed-off-by: Amir Goldstein <amir73il@users.sf.net>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

LorDClockaN committed:
ADD: ext4: dynamically allocate the jbd2_inode in ext4_inode_info as necessary

    Replace the jbd2_inode structure (which is 48 bytes) with a pointer
    and only allocate the jbd2_inode when it is needed --- that is, when
    the file system has a journal present and the inode has been opened
    for writing.  This allows us to further slim down the ext4_inode_info
    structure.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: reorder ext4_inode_info structure elements to remove unneeded padding

    By reordering the elements in the ext4_inode_info structure, we can
    reduce the padding needed on an x86_64 system by 16 bytes.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: drop i_state_flags on architectures with 64-bit longs

    We can store the dynamic inode state flags in the high bits of
    EXT4_I(inode)->i_flags, and eliminate i_state_flags.  This saves 8
    bytes from the size of ext4_inode_info structure, which when
    multiplied by the number of the number of in the inode cache, can save
    a lot of memory.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Clean up s_dirt handling

    We don't need to set s_dirt in most of the ext4 code when journaling
    is enabled.  In ext3/4 some of the summary statistics for # of free
    inodes, blocks, and directories are calculated from the per-block
    group statistics when the file system is mounted or unmounted.  As a
    result the superblock doesn't have to be updated, either via the
    journal or by setting s_dirt.  There are a few exceptions, most
    notably when resizing the file system, where the superblock needs to
    be modified --- and in that case it should be done as a journalled
    operation if possible, and s_dirt set only in no-journal mode.

    This patch will optimize out some unneeded disk writes when using ext4
    with a journal.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Convert more i_flags references to use accessor functions

    These changes are not ones which are likely to result in races, but
    they should be fixed.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: remove initialized but not read variables

    No real bugs found, just removed some dead code.

    Found by gcc 4.6's new warnings.

    Signed-off-by: Andi Kleen <ak@linux.intel.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: update ctime when changing the file's permission by setfacl

    ext4 didn't update the ctime of the file when its permission was
    changed.

    Steps to reproduce:
     # touch aaa
     # stat -c %Z aaa
     1275289822
     # setfacl -m  'u::x,g::x,o::x' aaa
     # stat -c %Z aaa
     1275289822                         <- unchanged

    But, according to the spec of the ctime, ext4 must update it.

    Port of ext3 patch by Miao Xie <miaox@cn.fujitsu.com>.

    CC: linux-ext4@vger.kernel.org
    Signed-off-by: Jan Kara <jack@suse.cz>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: clean up ext4_abort() so __func__ is now implicit

    Use a macro definition for ext4_abort() to clean up the .c files a wee
    bit.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Enhance ext4_grp_locked_error() to take block and function numbers

    Also use a macro definition so that __func__ and __LINE__ is implicit.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Pass line number to ext4_journal_abort_handle()

    This allows the error messages to include the line number

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Cleanup ext4_check_dir_entry so __func__ is now implicit

    Also start passing the line number to ext4_check_dir since we're going
    to need it in upcoming patch.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Pass line numbers to ext4_error() and friends

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Save error information to the superblock for analysis

    Save number of file system errors, and the time function name, line
    number, block number, and inode number of the first and most recent
    errors reported on the file system in the superblock.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Once a day, printk file system error information to dmesg

    This allows us to grab any file system error messages by scraping
    /var/log/messages.  This will make it easy for us to do error analysis
    across the very large number of machines as we deploy ext4 across the
    fleet.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Remove unnecessary casts of private_data

    Signed-off-by: Joe Perches <joe@perches.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Fix block bitmap inconsistencies after a crash when deleting files

    We have experienced bitmap inconsistencies after crash during file
    delete under heavy load.  The crash is not file system related and I
    the following patch in ext4_free_branches() fixes the recovery
    problem.

    If the transaction is restarted and there is a crash before the new
    transaction is committed, then after recovery, the blocks that this
    indirect block points to have been freed, but the indirect block
    itself has not been freed and may still point to some of the free
    blocks (because of the ext4_forget()).

    So ext4_forget() should be called inside ext4_free_blocks() to avoid
    this problem.

    Signed-off-by: Amir Goldstein <amir73il@users.sf.net>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: jbd2: Remove __GFP_NOFAIL from jbd2 layer

    __GFP_NOFAIL is going away, so add our own retry loop.  Also add
    jbd2__journal_start() and jbd2__journal_restart() which take a gfp
    mask, so that file systems can optionally (re)start transaction
    handles using GFP_KERNEL.  If they do this, then they need to be
    prepared to handle receiving an PTR_ERR(-ENOMEM) error, and be ready
    to reflect that error up to userspace.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Support discard requests when running in no-journal mode

    Issue discard request in ext4_free_blocks() when ext4 has no journal and
    is mounted with discard option.

    Signed-off-by: Jiaying Zhang <jiayingz@google.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Fix potential memory leak in ext4_fill_super

    Under heavy memory pressure we may hit out of memory
    situation and as result kstrdup'ed options will not be
    freed. Fix it.

    Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Always journal quota file modifications

    When journaled quota options are not specified, we do writes
    to quota files just in data=ordered mode. This actually causes
    warnings from JBD2 about dirty journaled buffer because ext4_getblk
    unconditionally treats a block allocated by it as metadata. Since
    quota actually is filesystem metadata, the easiest way to get rid
    of the warning is to always treat quota writes as metadata...

    Signed-off-by: Jan Kara <jack@suse.cz>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: fix ext4_get_blocks references

    ext4_get_blocks got renamed to ext4_map_blocks, but left stale
    comments and a prototype littered around.

    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
FIX: ext4: fix EFBIG edge case when writing to large non-extent file

    By running the following reproducer, we can confirm that the write
    system call returns with 0 when it should return the error EFBIG.

    #!/bin/sh

    /bin/dd if=/dev/zero of=./img bs=1k count=1 seek=1024k > /dev/null 2>&1
    /sbin/mkfs.ext3 -Fq ./img
    /bin/mount -o loop -t ext4 ./img /mnt
    /bin/touch /mnt/file
    strace /bin/dd if=/dev/zero of=/mnt/file conv=notrunc bs=1k count=1 seek=$((2194719883264/1024)) 2>&1 | /bin/egrep "write.* 1024\) = "
    /bin/umount /mnt
    exit

    Signed-off-by: Toshiyuki Okajima <toshi.okajima@jp.fujitsu.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
    Cc: Eric Sandeen <sandeen@redhat.com>

LorDClockaN committed:
FIX: ext4: don't print scary messages for allocation failures post-abort

    I often get emails containing the "This should not happen!!" message,
    conveniently trimmed to remove things like:

    sd 0:0:0:0: [sda] Unhandled error code
    sd 0:0:0:0: [sda] Result: hostbyte=DID_OK driverbyte=DRIVER_TIMEOUT
    sd 0:0:0:0: [sda] CDB: Write(10): 2a 00 03 13 c9 70 00 00 28 00
    end_request: I/O error, dev sda, sector 51628400
    Aborting journal on device dm-0-8.
    EXT4-fs error (device dm-0): ext4_journal_start_sb: Detected aborted journal
    EXT4-fs (dm-0): Remounting filesystem read-only

    I don't think there is any value to the verbosity if the reason is
    due to a filesystem abort; it just obfuscates the root cause.

    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
FIX: ext4: drop inode from orphan list if ext4_delete_inode() fails

    There were some error paths in ext4_delete_inode() which was not
    dropping the inode from the orphan list.  This could lead to a BUG_ON
    on umount when the orphan list is discovered to be non-empty.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: force block allocation on quota_off

    Perform full sync procedure so that any delayed allocation blocks are
    allocated so quota will be consistent.

    Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ADD: ext4: Add mount options in superblock

    Allow mount options to be stored in the superblock.  Also add default
    mount option bits for nobarrier, block_validity, discard, and nodelalloc.

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ext4: re-inline ext4_rec_len_(to|from)_disk functions

    commit 3d0518f4, "ext4: New rec_len encoding for very
    large blocksizes" made several changes to this path, but from
    a perf perspective, un-inlining ext4_rec_len_from_disk() seems
    most significant.  This function is called from ext4_check_dir_entry(),
    which on a file-creation workload is called extremely often.

    I tested this with bonnie:

    # bonnie++ -u root -s 0 -f -x 200 -d /mnt/test -n 32

    (this does 200 iterations) and got this for the file creations:

    ext4 stock:   Average =  21206.8 files/s
    ext4 inlined: Average =  22346.7 files/s  (+5%)

    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ext4: Fix dirtying of journalled buffers in data=journal mode

    In data=journal mode, we still use block_write_begin() to prepare
    page for writing. This function can occasionally mark buffer dirty
    which violates journalling assumptions - when a buffer is part of
    a transaction, it should be dirty and a buffer can be already part
    of a forget list of some transaction when block_write_begin()
    gets called. This violation of journalling assumptions then results
    in "JBD: Spotted dirty metadata buffer..." warnings.

    In fact, temporary dirtying the buffer while the page is still locked
    does not really cause problems to the journalling because we won't write
    the buffer until the page gets unlocked. So we just have to make sure
    to clear dirty bits before unlocking the page.

    Signed-off-by: Jan Kara <jack@suse.cz>

zachariasmaladroit committed:
ext4: Fix dirtying of journalled buffers in data=journal mode [other
patch by LorD ClockaN]

    In data=journal mode, we still use block_write_begin() to prepare
    page for writing. This function can occasionally mark buffer dirty
    which violates journalling assumptions - when a buffer is part of
    a transaction, it should be dirty and a buffer can be already part
    of a forget list of some transaction when block_write_begin()
    gets called. This violation of journalling assumptions then results
    in "JBD: Spotted dirty metadata buffer..." warnings.

    In fact, temporary dirtying the buffer while the page is still locked
    does not really cause problems to the journalling because we won't write
    the buffer until the page gets unlocked. So we just have to make sure
    to clear dirty bits before unlocking the page.

    Signed-off-by: Jan Kara <jack@suse.cz>

coolya committed:
Merge pull request #32 from jt1134/0f351295cbd193a99ad5845ee2fab7694e751bcc

fascinate: fix in-call audio

techmik committed:
cleanup =)

LorDClockaN committed:
ADD: convert logfs to ->evict_inode()

    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>

LorDClockaN committed:
BKL: Remove BKL from ext4 filesystem

    The BKL is still used in ext4_put_super(), ext4_fill_super() and
    ext4_remount(). All three calles are protected against concurrent calls by
    the s_umount rw semaphore of struct super_block.

    Therefore the BKL is protecting nothing in this case.

    Signed-off-by: Jan Blunck <jblunck@infradead.org>
    Acked-by: "Theodore Ts'o" <tytso@mit.edu>
    Signed-off-by: Arnd Bergmann <arnd@arndb.de>

LorDClockaN committed:
ext4: fix EOFBLOCKS_FL handling

    It turns out we have several problems with how EOFBLOCKS_FL is
    handled.  First of all, there was a fencepost error where we were not
    clearing the EOFBLOCKS_FL when fill in the last uninitialized block,
    but rather when we allocate the next block _after_ the uninitalized
    block.  Secondly we were not testing to see if we needed to clear the
    EOFBLOCKS_FL when writing to the file O_DIRECT or when were converting
    an uninitialized block (which is the most common case).

    Google-Bug-Id: 2928259

    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

LorDClockaN committed:
ext4: use dedicated slab caches for group_info structures

    ext4_group_info structures are currently allocated with kmalloc().
    With a typical 4K block size, these are 136 bytes each -- meaning
    they'll each consume a 256-byte slab object.  On a system with many
    ext4 large partitions, that's a lot of wasted kernel slab space.
    (E.g., a single 1TB partition will have about 8000 block groups, using
    about 2MB of slab, of which nearly 1MB is wasted.)

    This patch creates an array of slab pointers created as needed --
    depending on the superblock block size -- and uses these slabs to
    allocate the group info objects.

    Google-Bug-Id: 2980809

    Signed-off-by: Curt Wohlgemuth <curtw@google.com>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

coolya committed:
Revert "video: add new mfc driver from samsung source drop"

This reverts commit 78d6f3d4d8a32cd4ae27a20e786194f8c9b22f66.

coolya committed:
Merge branch 'android-samsung-2.6.35' of https://github.com/teamhacksung/android_kernel_samsung_aries into HEAD

atinm committed:
Revert "ce147: add missing autofocus mode"

This reverts commit 28bf4cf22f1cf63266e07896cd974a2785b717c1.

Conflicts:

	arch/arm/mach-s5pv210/cpuidle.c

Change-Id: I212c070a6ebde86133a3b4fd80310cadd23e3fb5

atinm committed:
Revert "ce147: update driver from latest samsung source drop"

This reverts commit 35e80ed772c09d0ff24c483bb79e0f754f0944a9.

atinm committed:
Merge branch 'bluetooth' of github.com:teamhacksung/android_kernel_samsung_aries into android-samsung-2.6.35

atinm committed:
Hack a mute/unmute for Vibrant - doesn't really help it seems

Change-Id: Icb4d2a3df1161dedc7da50c3e4cacf12a003f54a

jt1134 committed:
fascinate: port optical drivers from EB01 froyo source

Change-Id: I86c093866597572105b4a83a37af0109b55bd563

atinm committed:
Merge pull request #33 from techmik/android-samsung-2.6.35

cleanup =)

atinm committed:
Merge pull request #35 from jt1134/pull-request

fixes for fascinate

techmik committed:
fix permissions

Change-Id: I728c49ef2b4259edd67def9d5c25a72c89a63f34

atinm committed:
Merge pull request #36 from techmik/gingerbread

fix permissions

existz committed:
iosched: Add V(R) i/o scheduler

atinm committed:
proximity sensor: initialize to 'far'.
sec_jack: headphone detection at boot.
video: updated bootsplash.

Change-Id: I3b45b907dc5793ba9e8d4d8905b31362e5881c49

jt1134 committed:
fascinate: fix export of backlight_level in panel driver so that ambient light sensor will behave properly

Commit:
ashmem: fix deadlock when system is low on memory

When system is low on memory, allocating memory while holding the
ashmem_mutex may try to directly reclaim memory. Then ashmem_shrink()
is called in same thread. It will deadlock at acquiring ashmem_mutex.
This change lets ashmem_shrink() return failure if ashmem_mutex is not
available. Memory will be reclaimed from others.

Change-Id: I3b6024d5477d37245f4ebde6d7069fb77af789ee
Signed-off-by: Yuanyuan Zhong <yuanyuan.zhong@gmail.com>

jt1134 committed:
fascinate: GPIO_CP_RST fixes

jt1134 committed:
Merge pull request #37 from jt1134/pull-request

additional fixes for fascinate

jt1134 committed:
fascinate: fix issue where phone would not wakeup for incoming calls/msgs when in deep sleep

jt1134 committed:
fascinate: moar gpio fixes

existz committed:
s3cfb: port nexus s framebuffer driver; add voodoo color for 2.3.4

supercurio committed:
Voodoo sound: driver v10

Improvement:

- HW EQ support: smooth activation/deactivation and gain changes

Bugfixes:

- HP volume smoothing loop on low levels with negative digital offsets
- wm8994_write logging on Nexus S

New supported devices:

- Galaxy Tab 7" Gingerbread Kernels support (based on M180S)
- Galaxy Tab 10.1 - beta

zachariasmaladroit committed:
Revert "Turn off LOG_INFO" (wm8994 extensions / voodoo sound)

This reverts commit 9b05a4409b2acb34be43069d8a48fd0ea0f194bd.

zachariasmaladroit committed:
Revert "turn off extension debug"

This reverts commit 86efa843c75fbd5b18d1fb32e5672739fa14e704.

zachariasmaladroit committed:
Revert "rename wm8994 'voodoo sound' extensions to just wm8994_extensions."

This reverts commit 385396a7488e4a2d7f9a8c8f73ec01b472b6cef4.

existz committed:
Adjust colour multiplier to avoid yellow tint shock

atinm committed:
Merge pull request #38 from zachariasmaladroit/pmem_ashmem

Pmem ashmem

atinm committed:
Enabled BT PAN support. HID support

Change-Id: Iffc5f8d5d4958952b04cfdc892f1db7fdac606db

atinm committed:
Merge branch 'android-samsung-2.6.35' of git://github.com/teamhacksung/android_kernel_samsung_aries into android-samsung-2.6.35

existz committed:
s3cfb: add mDNIe from nexus s

jt1134 committed:
fascinate: use yamaha compass drivers from ED01

yath committed:
Enable CONFIG_TIMER_STATS

Set CONFIG_TIMER_STATS to 'y' so powertop will work.

atinm committed:
enabled BT_HID support, HID_APPLE support

Change-Id: I5fd1b70b0f80bf376463d1726643b24dc120c4dc

renaudallard committed:
Port BFQ V2 to 2.6.35

yath committed:
Build fuse.ko

Build FUSE module.

coolya committed:
Merge pull request #39 from yath/android-samsung-2.6.35

Enable CONFIG_TIMER_STATS

coolya committed:
Merge pull request #40 from yath/config_fuse

Build fuse.ko

renaudallard committed:
sdhci: add regulator support

This patch adds support for regulator API to sdhci core driver.
Regulators can be used to disable power in suspended state to reduce
dissipated energy.

Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Acked-by: Mark Brown <broonie@opensource.wolfsonmicro.com>
Cc: <linux-mmc@vger.kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Conflicts:

	drivers/mmc/host/sdhci.c

pawitp committed:
Updated camera driver from GT-I9000_OpenSource_GB_Update.tar.gz

Requires libcamera.so from newer ROMs.

eifert committed:
Setup MAX8998_LDO5 regulator on machine init and expose it for a "vmmc" consumer. Fixes broken external SD when using Samsung Gingerbread bootloader.

coolya committed:
Merge pull request #42 from eifert/android-samsung-2.6.35

External SD card fix for Gingerbread bootloader

jt1134 committed:
update gpio settings from ED05 source drop

jt1134 committed:
Merge branch 'android-samsung-2.6.35' of github.com:teamhacksung/android_kernel_samsung_aries into android-samsung-2.6.35

coolya committed:
Merge pull request #41 from pawitp/android-samsung-2.6.35

Updated camera driver from GT-I9000_OpenSource_GB_Update.tar.gz (Fixes front camera orientation).

existz committed:
Upgrade to 2.6.35.10

ytt3r committed:
readahead: change default values

ytt3r committed:
ram tweaks: 341mb free, 720p should still be working; needs testing

ytt3r committed:
Added an option to disable printk in the defconfig.

ytt3r committed:
Kconfig option to disable printk

ytt3r committed:
Added buildscript to build for all devices.

ytt3r committed:
Buildscript to compile and package TalonMTD in one command

ytt3r committed:
Compile the Conservative governor as a module

ytt3r committed:
Merge branch 'android-samsung-2.6.35' of github.com:ytt3r/android_kernel_samsung_aries into android-samsung-2.6.35

ytt3r committed:
Releasetools

ytt3r committed:
net: wireless: bcm4329: Reduce listen interval to 10

ytt3r committed:
cpufreq: set initial max frequency to 1GHz for boot security

ytt3r committed:
arm: Allow CPU-supported unaligned accesses

This change reconfigures the CPU to allow CPU-supported unaligned
accesses, which are generally faster than software-only fixups,
resulting in fewer alignment exceptions.

Signed-off-by: Brent DeGraaf <bdegraaf@codeaurora.org>

ytt3r committed:
lowmemorykiller.c: instead of killing the largest available task, kill the task closest in size to the amount of memory we actually need to free to satisfy the minfree threshold. (courtesy of kodos96)

zachariasmaladroit committed:
add support for telus fascinate - this should keep compatibility for the i9000 / SGS - needs testing

ytt3r committed:
Add a compiled initramfs and mkshbootimg.py for the compile script

ytt3r committed:
cpufreq: undervolt for stock frequencies

ytt3r committed:
Compile android logger (logcat) as a module, can be enabled with logcat-on.
Appropriate changes have been made to releastools and the buildscript.

ytt3r committed:
Removed obsolete buildscript and renamed the current one to build.sh.

ytt3r committed:
CFS: load balancing improvements
(Signed-off-on-by: Nikhil Rao; granularity improvements by Peter Ziljstra)

ytt3r committed:
Updated makefile for new toolchain

ytt3r committed:
Buildscript now copies all modules to releasetools

ytt3r committed:
1200MHz overclock

ytt3r committed:
Overclock up to 1440MHz following logical steps

zachariasmaladroit committed:
ext4: Automatically allocate delay allocated blocks before rename
Ext4 already performs block allocation on rename, but it does it after
rename was complete, so race window is much tighter but still exist.
In order to guarantee atomic behavior let's force block allocation before
actual rename. This rearrangement should not affect IO performance.

- Code cleanup: It is possible to move some checks out of transaction
  which theoretically may result in better performance.

Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>

zachariasmaladroit committed:
ext4: Fix overflow of metadata block reservation counter (Signed-off-by: Akira Fujita <a-fujita@rs.jp.nec.com>)

zachariasmaladroit committed:
ext4: allow inode_readahead_blks=0

ytt3r committed:
Enable AIO and SIO

ytt3r committed:
Kernel HZ to 250

ytt3r committed:
Aesthetics of updater-script

ytt3r committed:
Kangsterizer's LED notifications
Using as a replacement for BLN until I or someone else gets BLN working satisfactorily.

ytt3r committed:
ARM: aries: lower voltage for RAM(1.8v->1.6v) and ADC(3.3v->3.0v)

ytt3r committed:
Enabled BFQ2

ytt3r committed:
Disabled userspace

ytt3r committed:
Talon bootscreen

ytt3r committed:
Added smartass

existz committed:
Revert "CFS: load balancing improvements"

This reverts commit 1ac14a4eb62f966d1ec74962792b21053d90894a.

ytt3r committed:
Update recovery with Talon options

ytt3r committed:
Raise GPU max clock speed

ytt3r committed:
GPU tweaks from Morfic

ytt3r committed:
Added Scholastic Fair Blue network scheduler to defconfigs

ytt3r committed:
Removed redundant HZ value; Set the important one to 250.

ytt3r committed:
Revert "Added Scholastic Fair Blue network scheduler to defconfigs"

The scheduler was causing the device to hard reboot after toggling airplane mode on.

This reverts commit 4dacc92e63dc7c4e29ec49db24aa50debb3188fb.

ytt3r committed:
Enable conservative on every boot

ytt3r committed:
Add a missing file for nexus s framebuffer

ytt3r committed:
Renamed kernel to TalonMTD in version.c

ytt3r committed:
Update recovery again

ytt3r committed:
Revert toolchain to arm-eabi-4.4.3, Linaro was causing reboot issues

pawitp committed:
Added low power mode support

Change-Id: If41c7b37585210e71c08660bd716d4799b4b55da

ytt3r committed:
Revert "Renamed kernel to TalonMTD in version.c"

This reverts commit 248bd42ae70b9a5cd6b9622caa51641f23364f18.

ytt3r committed:
Disabled backlight timeout, still have to find a way to get the lights to come on with the screen.

ytt3r committed:
Fix LED screen off

ytt3r committed:
ARM: SAMSUNG: Fix on s5p_gpio_[get,set]_drvstr

This patch fixes bug on gpio drive strength helper function.

The offset should be like follwoing.
-       off = chip->chip.base - pin;
+       off = pin - chip->chip.base;

In the s5p_gpio_get_drvstr(),
the second line is unnecessary, because overwrite drvstr.
        drvstr = __raw_readl(reg);
-       drvstr = 0xffff & (0x3 << shift);

And need 2bit masking before return the drvstr value.
        drvstr = drvstr >> shift;
+       drvstr &= 0x3;

In the s5p_gpio_set_drvstr(), need relevant bit clear.
        tmp = __raw_readl(reg);
+       tmp &= ~(0x3 << shift);
        tmp |= drvstr << shift;

Reported-by: Jaecheol Lee <jc.lee@samsung.com>
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

ytt3r committed:
Enabled cache wipe and init.d cleaning script

ytt3r committed:
ARM: SAMSUNG: Fix on drive strength value

This patch fixes on defined drive strength value for GPIO.
According to data sheet, if we want drive strength 1x, the value
should be 00(b), if 2x should be 10(b), if 3x should be 01(b),
and if 4x should be 11(b). Also fixes comment(from S5C to S5P).

Reported-by: Janghyuck Kim <janghyuck.kim@samsung.com>
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

ytt3r committed:
ARM: SAMSUNG: MMC: fix build error when both DMA and PIO mode selected

[cjb: fix line-wrapped patch]
Signed-off-by: Jiri Pinkava <jiri.pinkava@vscht.cz>
Signed-off-by: Chris Ball <cjb@laptop.org>
Cc: Matt Fleming <matt@console-pimps.org>
Cc: Russell King <rmk@arm.linux.org.uk>
Cc: Ben Dooks <ben-linux@fluff.org>
Cc: <linux-mmc@vger.kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

ytt3r committed:
ARM: S5P: Fix DMA coherent mask for FIMC

FIMC driver uses DMA_coherent allocator, which requires proper dma mask
to be set.

Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
[kgene.kim@samsung.com: minor title fix]
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

ytt3r committed:
ARM: S5P: Decrease IO Registers memory region size on FIMC

IO registers region size of all FIMC versions is less than 1kB so there
is no need to reserve 1M.

Signed-off-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
[kgene.kim@samsung.com: minor title fix]
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

ytt3r committed:
ARM: S5PV210: Update I2C-1 Clock Register Property.

CLK_GATE_IP3[8] is RESERVED. The port "I2C_HDMI_DDC" of CLK_GATE_IP3[10] is
used as another I2C port. Therefore, defined the unused I2C-1 as another I2C
there was left undefined but used.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

ytt3r committed:
ARM: S5PV210: Add FIMC clocks

These clocks enables FIMC driver to operate on machines, which
bootloader power gated FIMC devices to save power on boot.

Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
[kgene.kim@samsung.com: minor title fix]
Signed-off-by: Kukjin Kim <kgene.kim@samsung.com>

pawitp committed:
Ported accidental touch key prevention from stock kernel.

Try sliding down your touch screen onto your menu or back key. This
prevents the capacitive keys from being accidentally triggered

Change-Id: I47dc3e8ae869b422cc092d2a51db6a2e12cdebad

pawitp committed:
Enabled DVFS_LIMIT for MFC to reduce 720p stutter (ported from stock)

Change-Id: Ia5db1f2cec9c4634253bd6a7199d4a56fbb86c80

ytt3r committed:
Fixed dvfslock to work with overclock settings

ytt3r committed:
cpufreq: some tweaks to scale up and down more steadily instead of jumping directly to frequencies.

ytt3r committed:
Fixed enable_conservative

ytt3r committed:
cpufreq: added ondemandx governor

jpjust committed:
Changed CONFIG_INPUT_YAS529_POSITION to 0 (magnetometer position copied from Samsung's GT-I9000B kernel config) and added CONFIG_INPUT_YAMAHA. Compass works now after proper calibration.

atinm committed:
Only set batt_max_soc if the batt_soc reported is greater than previous

Change-Id: I501c3feb233088247e9dfe52cc117d00b0b705cc

pawitp committed:
Change battery percentage calculation.

Compensate for chipset inaccuracies in returning battery percentage
the way Samsung did it in its Froyo rom. This allows the battery to
reach 100% (may require bump charging) and no visible battery drop
is noticable (even after a reboot). The precision is also increased
(no 2% jump) from the current method of calculation because the
calculation is done where we have extra precision information from
the chipset.

Change-Id: Id7f15dd3906f7a12949683c0d921ce7a8498e70a

ytt3r committed:
Fixed ondemandx

ytt3r committed:
Bumped OC voltages; people were reporting overclock instability with the current voltages.

ytt3r committed:
ARM: S5PC11X: improve USB signal quality

Ported from AOSP Samsung source - http://android.git.kernel.org/?p=kernel/samsung.git;a=summary

Tunning of PHY0 Tune Register to improve physical characteristics.
This improves the margin of USB eye diagram, which means
the quality of USB signal is also improved.

ytt3r committed:
Fix compile error introduced in the last commit

ytt3r committed:
Updated version to 0.0.2 (a little late :P)

ytt3r committed:
Renamed logcat scripts to 'logcat-on' and 'logcat-off'

ytt3r committed:
Updated initramfs to support low power mode

renaudallard committed:
There is no NFC in the SGS

jt1134 committed:
boost in-call volumes

credit to efpophis

ytt3r committed:
Upped version to 0.0.3.

ytt3r committed:
Changed script to fit Teamhacksung's device naming conventions (captivate -> captivatemtd)

erasmux committed:
smartassV2 governor (backport)

pawitp committed:
Protect cypress i2c reads/writes with a mutex

This should fix a rare race condition causing the touchkeys to fail
(due to bl_off timer) as i2c isn't thread-safe. When the touchkeys
fails this way, the screen won't turn on and this may be certain
cases of the so called "sleep of death".

Change-Id: Id9b0af983c17a562058edf0154a171558a147584

ytt3r committed:
cpufreq: smartass tweaks
Defconfigs updated to include smartass 2

ytt3r committed:
Disable KTLO for everyone except Captivatemtd, users were reporting it was just an annoyance.

ytt3r committed:
Added a Talon-specific readme.

ytt3r committed:
Fixed errors related to building with the build script;
Enabled easy setup of cross compiler from script.

ytt3r committed:
Updated README.talon and Makefile to reflect the new cross compile style

ytt3r committed:
Updated wm8994_herring.c for louder volumes courtesy of hardcore

ytt3r committed:
Introduced voltages.h, a simple consolidation of CPU voltages.

ytt3r committed:
Made Smartass more battery efficient in screen off

pawitp committed:
Fix media buttons on Galaxy S

Samsung's source for aries doesn't set ear_sel from wm8994.

Change-Id: I495f4c8342e3d589b12fe42dbdb369f55f9c7545

ytt3r committed:
cpufreq: smartass2: changed ideal screen on freq to 600

pawitp committed:
Cleaned up media key patch and enabled for all variants.

- CAPTIVATE ifdef in wm8994 drivers are removed because they no longer
  have any effect

Change-Id: I8e26ecc4df63a813a665ab012cd7c6c49651c061

pawitp committed:
Increase ADC range for media key on Galaxy S to match Samsung stock

Change-Id: Ib1c8e3cd98046477ff06de98089264f42c2c560d

ytt3r committed:
Fixed GPU Overclock

ytt3r committed:
Reset any smartass commits

ytt3r committed:
Merge branch 'android-samsung-2.6.35' of existz.darkirc.info:kernel/talonmtd into android-samsung-2.6.35

zachariasmaladroit committed:
updated initramfs files to state of #124 (captivate + galaxys/gt-i9000 counting scheme)

zachariasmaladroit committed:
Merge branch 'android-samsung-2.6.35' of existz.darkirc.info:kernel/talonmtd into android-samsung-2.6.35

Commit:
sched: Improve latencies under load by decreasing minimum scheduling granularity

Mathieu reported bad latencies with make -j10 kind of kbuild
workloads - which is mostly caused by us scheduling with a
too coarse granularity.

Reduce the minimum granularity some more, to make sure we
can meet the latency target.

I got the following results (make -j10 kbuild load, average of 3
runs):

 vanilla:

  maximum latency: 38278.9 µs
  average latency:  7730.1 µs

 patched:

  maximum latency: 22702.1 µs
  average latency:  6684.8 µs

Mathieu also measured it:

|
| * wakeup-latency.c (SIGEV_THREAD) with make -j10
|
| - Mainline 2.6.35.2 kernel
|
| maximum latency: 45762.1 µs
| average latency: 7348.6 µs
|
| - With only Peter's smaller min_gran (shown below):
|
| maximum latency: 29100.6 µs
| average latency: 6684.1 µs
|

Reported-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Reported-by: Linus Torvalds <torvalds@linux-foundation.org>
Acked-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Suggested-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
Acked-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
LKML-Reference: <AANLkTi=8m4g01wZPacySoF7U0PevTNVgJoZZrHiUD-pN@mail.gmail.com>
Signed-off-by: Ingo Molnar <mingo@elte.hu>

zachariasmaladroit committed:
initramfs [SGS - should also work with captivate] from October 9th (nightly #143/#144)
